kogitoq/kogito-common/src/main/java/life/genny/kogito/common/messages/SendAllMessages.java

```diff
@@ -0,0 +1,150 @@
+package life.genny.kogito.common.messages;
+
+import life.genny.qwandaq.attribute.EntityAttribute;
+import life.genny.qwandaq.entity.BaseEntity;
+import life.genny.qwandaq.entity.SearchEntity;
+import life.genny.qwandaq.models.UserToken;
+import life.genny.qwandaq.utils.BaseEntityUtils;
+import life.genny.qwandaq.utils.SearchUtils;
+import org.jboss.logging.Logger;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SendAllMessages extends MessageSendingStrategy {
+
+    @Inject
+    SearchUtils searchUtils;
+
+    private String productCode;
+    private String milestoneCode;
+    private BaseEntity coreBE;
+
+    public SendAllMessages(String productCode, String milestoneCode, BaseEntity coreBE) {
+        this.productCode = productCode;
+        this.milestoneCode = milestoneCode;
+        this.coreBE = coreBE;
+    }
+
+    public SendAllMessages(String milestoneCode, String coreBeCode) {
+        this.milestoneCode = milestoneCode;
+        this.coreBE = beUtils.getBaseEntity(userToken.getProductCode(), coreBeCode);
+        this.productCode = this.coreBE.getRealm();
+    }
+
+    @Override
+    public void sendMessage() {
+        SearchEntity searchEntity = new SearchEntity("SBE_MILESTONE_MESSAGES",
+                "Fetch All Messages associated with milestone Code")
+                .addFilter("PRI_CODE", SearchEntity.StringFilter.LIKE, "MSG_%")
+                .addFilter("PRI_MILESTONE", SearchEntity.StringFilter.LIKE, "%\"" + milestoneCode.toUpperCase() + "\"%")
+                .setPageStart(0)
+                .setPageSize(100);
+
+        searchEntity.setRealm(productCode);
+
+        List<String> messageCodes = searchUtils.searchBaseEntityCodes(searchEntity);
```

Comment:
Using Java parallel streams would be ideal since the processing of each messageCode is independent.

Reply:
updated.